recipes/spdlog/all/conanfile.py

```diff
@@ -41,10 +41,11 @@ def config_options(self):
             del self.options.fPIC
 
     def configure(self):
-        if self.options.shared or self.options.header_only:
-            self.options.rm_safe("fPIC")
         if self.options.header_only:
             del self.options.shared
+            self.options.rm_safe("fPIC")
+        elif self.options.shared:
+            self.options.rm_safe("fPIC")
```

Comment:
```suggestion
        if self.options.get_safe("shared") or self.options.header_only:
            self.options.rm_safe("fPIC")
        if self.options.header_only:
            self.options.rm_safe("shared")
```

But honestly it's a conan client bug for me.